third_party/amd/lib/TritonAMDGPUTransforms/StreamPipelineV2.cpp

```diff
@@ -52,6 +52,24 @@ static Operation *streamPredication(RewriterBase &rewriter, Operation *op,
   return tt::predicateOp(rewriter, op, pred);
 }
 
+void changeTransposeMemoryToMutable(OpBuilder &builder, Value newSource,
```

Comment:
note that there is a helper `replaceUsesAndPropagateType` that may be useful to not have to do that

Reply:
Thanks for pointing this out :)